webapp/src/app/services/phaser/helper.ts

```diff
@@ -72,31 +77,33 @@ export class Helper {
 		return JSON.parse(JSON.stringify(obj));
 	}
 	
-	public static getJson(key: string, callback: (json: any) => void) {
+	public static async getJson<T>(key: string): Promise<T | undefined> {
+		const jsonKey = key + '.json';
 		const scene = Globals.scene;
 		
-		// get json from cache
-		if (scene.cache.json.has(key)) {
-			return callback(scene.cache.json.get(key));
+		if (scene.cache.json.has(jsonKey)) {
+			return scene.cache.json.get(jsonKey);
+		}
+		
+		if (this.cache[jsonKey]) {
+			return this.cache[jsonKey];
```

Comment:
At this point, we might as well do everything ourselves. We just need to replace the instances of `json.get(` with the helper function.
```ts
const jsonKey = key + '.json';

if (this.cache[jsonKey]) {
	return this.cache[jsonKey];
}

const promise: Promise<T | undefined> = firstValueFrom(
	Globals.httpService.resolveFile(jsonKey)
		.pipe(
			map(file => JSON.parse(file)),
			catchError(() => {
				console.warn(`Failed to resolve resource: ${jsonKey}`);
				return of(undefined);
			})
		)
	);

this.cache[jsonKey] = promise;
return promise;
```